woocommerce/woocommerce-android
# Issue/2630 display grouped products

Part 1/3 of #2630. This PR adds logic to display products associated with grouped products.

#### Changes
- Added logic to display the product count for a grouped product, if the count > 0.
- Created a new fragment, module and View Model to handle the changes needed to show only the products for a specific grouped product.
- Split the `ProductListViewHolder` and `ProductListDiffUtil` to separate classes so that they can be reused by `GroupedProductListAdapter` as well as `ProductListAdapter`

#### Screenshots
<img src="https://user-images.githubusercontent.com/22608780/90363796-e9225b00-e080-11ea-907c-52b25781f17c.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/90363801-eaec1e80-e080-11ea-8ffa-dcb06e289c86.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/90363810-ecb5e200-e080-11ea-9af2-63de5c973b40.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/90363813-ee7fa580-e080-11ea-8b7e-af5ac392e90c.png" width="300"/>


#### Testing
- Click on a grouped product with atleast 2 associated products.
- Notice the grouped products section in the Product Detail screen.
- Click on the grouped products section and notice the list of products associated with this product, displayed correctly.
- Click on the `delete` icon for one of the items and verify:
  - that the "Done" menu button is displayed correctly.
  - that clicking on the back button displays the discard dialog.
- Click on the "Done" menu button and notice that 
  - the "Update" button is displayed correctly in the Product detail screen.
  - the grouped products section displays the correct count.
- Click on the "Update" menu button and verify that the product is updated correctly.

#### Notes
- ~This PR is in draft till this [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1668) can be reviewed and merged.~
- The option to add a grouped product will take place in a subsequent PR.

Update release notes:
